src/test/java/org/apache/sysds/test/functions/nativ/ImgUtilsTest.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sysds.test.functions.nativ;
+
+import org.apache.sysds.utils.ImgNativeHelper;
+import org.apache.sysds.utils.NativeHelper;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class ImgUtilsTest {
+
+    /*static {
+        System.loadLibrary("systemds_openblas-Darwin-x86_64");
```

Comment:
I would simply add such a static block inside the ImgNativeHelper.java that loads the correctly compiled file depending on the OS in use, if there is no such file error out.
The reason the Native loader does not work on Mac is simply because we have not yet compiled the files for this.

Reply:
Thank you, 
while adding the static block, can I assume that the java.library.path is already configured ?

Reply:
but still, we need to compile the new implementations in order to include them into the native library. Or should I push my own compiled lib for test ?

Reply:
No we compile it on each test run for that machine. You can find the appropriate code in our GitHub action script.